ledger/block/src/header/mod.rs

```diff
@@ -72,29 +74,27 @@ impl<N: Network> Header<N> {
             metadata,
         };
         // Ensure the header is valid.
-        match header.is_valid() {
-            true => Ok(header),
-            false => bail!("Invalid block header: {:?}", header),
-        }
+        header.check_validity().with_context(|| "Invalid block header")?;
+        Ok(header)
     }
 
     /// Returns `true` if the block header is well-formed.
-    pub fn is_valid(&self) -> bool {
-        match self.height() == 0u32 {
-            true => self.is_genesis(),
-            false => {
-                // Ensure the previous ledger root is nonzero.
-                *self.previous_state_root != Field::zero()
-                    // Ensure the transactions root is nonzero.
-                    && self.transactions_root != Field::zero()
-                    // Ensure the finalize root is nonzero.
-                    && self.finalize_root != Field::zero()
-                    // Ensure the ratifications root is nonzero.
-                    && self.ratifications_root != Field::zero()
-                    // Ensure the metadata is valid.
-                    && self.metadata.is_valid()
+    pub fn check_validity(&self) -> Result<()> {
```

Comment:
Same, why go through the effort of changing?

Reply:
Same answer as to the other one. I kept `is_valid` for backwards compatibility.